aqt/generate_combinations.py

```diff
@@ -0,0 +1,391 @@
+            else ("latest", "5.13.2", "5.9.9")
+        )
+        for version in versions:
+            if version == "5.9.9" and archive_id.extension == "wasm":
+                continue
+            for arch_name in MetadataFactory(
+                archive_id, architectures_ver=version
+            ).getList():
+                yield {
+                    "os_name": archive_id.host,
+                    "target": archive_id.target,
+                    "arch": arch_name,
+                }
+
+
+def iter_tool_variants() -> Generator[dict, None, None]:
+    for archive_id in iter_archive_ids(categories=("tools",)):
+        print("Fetching tool variants for {}".format(archive_id))
+        for tool_name in tqdm(sorted(MetadataFactory(archive_id).getList())):
+            if is_blacklisted_tool(tool_name):
+                continue
+            for tool_variant in MetadataFactory(
+                archive_id, tool_name=tool_name
+            ).getList():
+                yield {
+                    "os_name": archive_id.host,
+                    "target": archive_id.target,
+                    "tool_name": tool_name,
+                    "arch": tool_variant,
+                }
+
+
+def iter_qt_minor_groups(
+    host: str = "linux", target: str = "desktop"
+) -> Generator[Tuple[int, int], None, None]:
+    for cat in (
+        "qt5",
+        "qt6",
+    ):
+        versions: Versions = MetadataFactory(ArchiveId(cat, host, target)).getList()
+        for minor_group in versions:
+            v = minor_group[0]
+            yield v.major, v.minor
+
+
+def iter_modules_for_qt_minor_groups(
+    host: str = "linux", target: str = "desktop"
+) -> Generator[Dict, None, None]:
+    print("Fetching qt modules for {}/{}".format(host, target))
+    for major, minor in tqdm(list(iter_qt_minor_groups(host, target))):
+        cat = f"qt{major}"
+        yield {
+            "qt_version": f"{major}.{minor}",
+            "modules": MetadataFactory(
+                ArchiveId(cat, host, target), modules_ver=f"{major}.{minor}.0"
+            ).getList(),
+        }
+
+
+def list_qt_versions(host: str = "linux", target: str = "desktop") -> List[str]:
+    all_versions = list()
+    for cat in (
+        "qt5",
+        "qt6",
+    ):
+        versions: Versions = MetadataFactory(ArchiveId(cat, host, target)).getList()
+        for minor_group in versions:
+            all_versions.extend([str(ver) for ver in minor_group])
+    return all_versions
+
+
+def merge_records(arch_records) -> List[Dict]:
+    all_records: List[Dict] = []
+    hashes = set()
+    for record in arch_records:
+        _hash = record["os_name"], record["target"], record["arch"]
+        if _hash not in hashes:
+            all_records.append(record)
+            hashes.add(_hash)
+    for sorting_key in ("arch", "target", "os_name"):
+        all_records = sorted(all_records, key=lambda d: d[sorting_key])
+    return all_records
+
+
+def list_new_archive_versions() -> List[str]:
```

Comment:
new_archive (= old versions, offline installer , not supported) should not be into supported combo.

Reply:
removed